### PR TITLE
Sync the VIP spec with v41 of the 1622.2 standard.

### PR DESCRIPTION
**Pull request by jdmgoogle**
This syncs the draft 5.0 spec with v41 of the 1622.2 standard. There were some small changes around element and attribute names that I made in order to minimize the differences between the two. There will probably be one or two small changes in the coming days, but this should be the largest batch of changes.

@jungshadow @pstenbjorn and @cjerdonek Please review.

### CODE REVIEW

File: vip_spec.xsd [hunk 1]
```diff
@@ -189,7 +190,8 @@ reference elements for polling location
           <xs:complexType>
             <xs:sequence>
               <xs:element name="OrderedContest" type="OrderedContestType" minOccurs="0" maxOccurs="unbounded" />
-              <xs:element name="InfoUrl" type="xs:anyURI" minOccurs="0" />
+              <xs:element name="PartyId" type="xs:IDREF" minOccurs="0" maxOccurs="unbounded" />
```
Comment (cjerdonek) [inline vip_spec.xsd]: Casing: `PartyID`?

File: vip_spec.xsd [hunk 2]
```diff
@@ -8,6 +8,7 @@ reference elements for polling location
   <xs:element name="VipObject">
     <xs:complexType>
       <xs:choice maxOccurs="unbounded">
+
```
Comment (cjerdonek) [inline vip_spec.xsd]: Are the random empty lines being added a style thing?  We should probably document that if so.

**Comment by jdmgoogle** [comment]
Thanks. 1622 is using SomeId instead of SomeID. Do people have thoughts about changing to use that format?

File: vip_spec.xsd (hunk 2 above)
Comment (jdmgoogle) [inline vip_spec.xsd]: I added the lines around the 'choice' tags because the end of the file was kind of a mess. The other blank lines I added were to create consistency with the other definitions which had blank lines between them. Although yes, we should add blank line use to the style guide.

File: vip_spec.xsd [hunk 3]
```diff
@@ -244,11 +249,11 @@ reference elements for polling location
         <xs:element name="Contact">
           <xs:complexType>
             <xs:sequence>
-              <xs:element name="Url" type="xs:anyURI" minOccurs="0" maxOccurs="unbounded" />
+              <xs:element name="Uri" type="xs:anyURI" minOccurs="0" maxOccurs="unbounded" />
               <xs:element name="AddressLine" type="xs:string" minOccurs="0" maxOccurs="unbounded" />
               <xs:element name="Email" type="xs:string" minOccurs="0" maxOccurs="unbounded" />
               <xs:element name="Fax" type="xs:string" minOccurs="0" maxOccurs="unbounded" />
-              <xs:element name="HoursOpenID" type="xs:IDREF" minOccurs="0" />
+              <xs:element name="Schedule" type="xs:IDREF" minOccurs="0" />
```
Comment (cjerdonek) [inline vip_spec.xsd]: It seems like this should be `HoursOpenID` to match our style guide?  See [here](https://github.com/votinginfoproject/vip-specification/blob/vip5/docs/style_guide.md#ids).

File: vip_spec.xsd [hunk 4]
```diff
@@ -260,13 +265,14 @@ reference elements for polling location
...
             <xs:sequence>
               <xs:element name="ElectoralDistrictID" type="xs:IDREF" />
-              <xs:element name="ContactID" type="xs:IDREF" minOccurs="0" maxOccurs="unbounded" />
               <xs:element name="Name" type="InternationalizedText" />
               <xs:element name="ExternalIdentifierID" type="xs:IDREF" minOccurs="0" maxOccurs="unbounded" />
-              <xs:element name="Partisan" type="xs:boolean" minOccurs="0" />
-              <xs:element name="TermEndDate" type="xs:date" minOccurs="0" />
-              <xs:element name="TermStartDate" type="xs:date" minOccurs="0" />
-              <xs:element name="TermType" type="TermType" />
+              <xs:element name="OfficeHolder" type="xs:IDREF" minOccurs="0" maxOccurs="unbounded" />
```
Comment (cjerdonek) [inline vip_spec.xsd]: Again, our style is to end in `[TypeName]ID` for ID references.  (By the way, I do think the guideline helps because it says what type the ID is a reference to.)

**Comment by cjerdonek** [comment]
> Do people have thoughts about changing to use that format?

I'd be okay with either.

**Comment by pstenbjorn** [comment]
I am happy with either, visually I prefer `[TypeName]ID` but `[TypeName]Id` seems to be what 1622.2 is standardizing around.

**Comment by jdmgoogle** [comment]
Okay, I've made the recommended updates. Once this PR goes in I'm going to do another round of changes to the XSD and the style guide to change things like ReferenceToID to ReferenceToId just to be consistent with 1622.

**Comment by jdmgoogle** [comment]
So 1622.2 added the concept of the current office holder to the Office object (there was some misunderstanding in what I was asking for, but I actually like it). The problem is that if we just use PersonID it's ambiguous as to which person this means. Suggestions?

**Comment by cjerdonek** [comment]
I would prefix it then, perhaps "OfficeHolderPersonID".  The style guide anticipated this by saying that the  attribute name should end with the type name followed by "ID".

File: vip_spec.xsd [hunk 5]
```diff
@@ -200,6 +202,7 @@ reference elements for polling location
             <xs:sequence>
               <xs:element name="PartyID" type="xs:IDREF" minOccurs="0" />
               <xs:element name="PersonID" type="xs:IDREF" minOccurs="0" />
+              <!-- SURVEY: Should external identifiers be referenced or included inline? -->
```
Comment (cjerdonek) [inline vip_spec.xsd]: Personally, I think inline makes more sense.  I also think we should define an "ExternalIdentifiers" (or "ExternalIdentifierCollection") type since everywhere we want to allow external identifiers, it will always be a collection.

Comment (jdmgoogle) [inline vip_spec.xsd]: Okay. I think that should be a different PR, but I just wanted to get everyone's opinion here.

**Comment by jdmgoogle** [comment]
Once @pstenbjorn and @jungshadow sign off on this, I'm going to merge this PR.

File: vip_spec.xsd (hunk 5 above)
Comment (cjerdonek) [inline vip_spec.xsd]: Cool, do you want to create the issue?

**Comment by pstenbjorn** [comment]
@jdmgoogle I am fine with the changes and I agree that the ExternalIdentifiersCollection concept should be handled in a separate PR.

**Comment by jungshadow** [comment]
@jdmgoogle I'd like to look at this, but I also don't want to unreasonably hold this up (i.e. it may still take me a few hours to get to as I'm not in the office). If you have @cjerdonek's and @pstenbjorn's approval on this, probably safe to say it's solid and any niggling bits I might bring up won't be show-stoppers and potentially fixed in subsequent updates.

**Comment by jdmgoogle** [comment]
Okay. Merging and moving on to the next issue.

**Comment by pstenbjorn** [comment]
@jdmgoogle would you like me to take a first stab at Venetian Blind conversion next?

**Comment by jdmgoogle** [comment]
@pstenbjorn Let me make one or two other small changes and then yes, feel free to take a stab at the VB conversion.
